src/main/java/org/spongepowered/common/data/builder/tileentity/AbstractTileEntityBuilder.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.builder.tileentity;
+
+import com.google.common.base.Preconditions;
+import net.minecraft.block.Block;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.math.BlockPos;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.block.BlockType;
+import org.spongepowered.api.block.entity.BlockEntity;
+import org.spongepowered.api.block.entity.BlockEntityType;
+import org.spongepowered.api.data.persistence.*;
+import org.spongepowered.api.world.server.ServerWorld;
+import org.spongepowered.common.accessor.tileentity.TileEntityTypeAccessor;
+import org.spongepowered.common.util.Constants;
+
+import java.util.Optional;
+import java.util.Set;
+
+public abstract class AbstractTileEntityBuilder<T extends BlockEntity> extends AbstractDataBuilder<T> implements DataBuilder<T> {
+    protected AbstractTileEntityBuilder(Class<T> requiredClass, int supportedVersion) {
+        super(requiredClass, supportedVersion);
+    }
+
+    @Override
+    protected Optional<T> buildContent(DataView container) throws InvalidDataException {
+        Preconditions.checkNotNull(container);
+
+        if (!container.contains(Constants.TileEntity.TILE_TYPE, Constants.TileEntity.WORLD, Queries.POSITION_X, Queries.POSITION_Y, Queries.POSITION_Z)) {
+            return Optional.empty();
+        }
+
+        String worldName = container.getString(Constants.TileEntity.WORLD).get();
+        Optional<ServerWorld> worldOptional = Sponge.getGame().getServer().getWorldManager().getWorld(worldName);
```

Comment:
.orElseNull, no reason to fool with the Optional past this point imo.

Reply:
Just turn it into a `final ServerWorld serverWorld = Sponge.getGame().getServer().getWorldManager().getWorld(worldName).orElseThrow(() -> new InvalidDataException("...."));`